cmasher/utils.py

```diff
@@ -233,6 +239,102 @@ def _get_cmap_perceptual_rank(
+        n_rgb_levels (int, optional): Number of RGB levels for each colormap
+            segment. Defaults to 256.
+
+    Returns:
+        Colormap: The composite colormap.
+
+    Raises:
+        TypeError: If the list contains mixed datatypes or invalid
+            colormap names.
+        ValueError: If the cmaps contain only one single colormap,
+            or if the number of nodes is not one less than the number
+            of colormaps, or if the nodes do not contain incrementing values
+            between 0.0 and 1.0.
+
+    Note:
+        The colormaps are combined from bottom to top.
+
+    References:
+        - https://stackoverflow.com/questions/31051488/combining-two-matplotlib-colormaps/31052741#31052741
+
+    Examples:
+        # Example 1: Using colormap names
+        >>> cmaps = ["ocean", "prism", "coolwarm"]
+        >>> nodes = [0.2, 0.75]
+        >>> custom_cmap_1 = combine_cmaps(cmaps, nodes)
+
+        # Example 2: Using predefined colormaps
+        >>> cmap_0 = plt.get_cmap("Blues")
+        >>> cmap_1 = plt.get_cmap("Oranges")
+        >>> cmap_2 = plt.get_cmap("Greens")
+        >>> custom_cmap_2 = combine_cmaps([cmap_0, cmap_1, cmap_2])
+
+    """
+    # Check colormap datatype and convert to list[Colormap]
+    if len(cmaps) == 1:
+        raise ValueError("Cannot combine a single colormap.")
+    for cm in cmaps:
+        if not isinstance(cm, (Colormap, str)):
+            raise TypeError(f"Unsupported colormap type: {type(cm)}.")
+    _cmaps: list[Colormap] = [
+        cm if isinstance(cm, Colormap) else mpl.colormaps[cm] for cm in cmaps
+    ]
+
+    # Generate default nodes for equal separation
+    if nodes is None:
+        nodes = np.linspace(0, 1, len(_cmaps) + 1)[1:-1].tolist()
+
+    # Check nodes length
+    if len(nodes) != len(_cmaps) - 1:
+        raise ValueError(
+            "Number of nodes should be one less than the number of colormaps."
+        )
+
+    # Preprocess nodes list for easier segmentation
+    nodes.insert(0, 0.0)
+    nodes.append(1.0)
+
+    # Check node values
+    if not all(0.0 <= node <= 1.0 for node in nodes) or any(
+        nodes[i] >= nodes[i + 1] for i in range(len(nodes) - 1)
+    ):
+        raise ValueError(
+            "Nodes should contain incrementing values between 0.0 and 1.0."
+        )
+
+    # Generate composite colormap
+    combined_cmap_segments = []
+
+    for i, cmap in enumerate(_cmaps):
+        start_position = nodes[i]
+        end_position = nodes[i + 1]
+
+        # Calculate the length of the segment
+        segment_length = int(n_rgb_levels * (end_position - start_position))
+
+        # Append the segment to the combined colormap segments
+        combined_cmap_segments.append(cmap(np.linspace(0, 1, segment_length)))
+
+    # Combine the segments (from bottom to top)
+    return LinearSegmentedColormap.from_list(
+        "combined_cmap", np.vstack(combined_cmap_segments)
```

Comment:
It's good to have a default name here, but we should probably also expose it to users

Reply:
Really good point here, I should set it as a default argument. Thanks a lot for pointing out.